kernel/common/enumerable.rb

```diff
@@ -136,18 +130,18 @@ def group_by
     h
   end
 
-  def slice_before(arg = undefined, &block)
-    if block_given?
-      has_init = !(undefined.equal? arg)
-    else
-      raise ArgumentError, "wrong number of arguments (0 for 1)" if undefined.equal? arg
-      block = Proc.new{ |elem| arg === elem }
-    end
+  def slice_before(pattern = undefined, &block)
+    pattern_given = !(undefined.equal? pattern)
+
+    raise ArgumentError, "cannot pass both pattern and block" if pattern_given && block_given?
+    raise ArgumentError, "wrong number of arguments (0 for 1)" if !pattern_given && !block_given?
```

Comment:
@jsyeo what about `"a pattern or a block must be provided"` for this to explain the error better?

Also, I've added you to the Rubinius contributors. Feel free to merge this and new commits if you feel confident in them. If you'd like feedback or are unsure, please use a PR.

Thanks!